plutus-wallet-api/src/Language/PlutusTx/StateMachine.hs

```diff
@@ -29,53 +30,62 @@ import           Ledger.Validation         (PendingTx, findContinuingOutputs, fi
 -- next state from the current state and an input, and a checking function that checks the validity
 -- of the transition in the context of the current transaction.
 data StateMachine s i = StateMachine {
+      -- | The transition function of the state machine. 'Nothing' indicates an invalid transition from the current state.
       smTransition :: s -> i -> Maybe s,
-      smCheck :: s -> i -> PendingTx -> Bool
+      -- | The condition checking function. Checks whether a given state transition is allowed given the 'PendingTx'.
+      smCheck :: s -> i -> PendingTx -> Bool,
+      -- | The final state predicate. Indicates whether a given state is final (the machine halts in that state).
+      smFinal :: s -> Bool
```

Comment:
Just a thought - maybe this should be `continuingOutputs :: s -> Int`, and then we'd look at the length of the list returned by `findContinuingOutputs ptx` in `mkValidator`. I know we don't currently handle splits in the state machine but this could be useful if we ever decide to allow them.

Reply:
Hm, maybe, but I'd prefer to change that if/when we handle splits, rather than trying to guess now what we'll want.

Reply:
Hm maybe, but I'd prefer to handle that if/when we handle splits rather than trying to guess what we'll want.

Reply:
ok 👍